src/debug/javaDebugProvider.ts

```diff
@@ -107,4 +104,45 @@ export class JavaDebugProvider implements IDebugProvider {
             appPort: Number(rawAppPortInfo)
         };
     }
+
+    public async resolvePortsFromContainer(kubectl: Kubectl, pod: string, container: string): Promise<PortInfo> {
+        let rawDebugPortInfo: string;
+
+        const execCmd = `exec ${pod} ${container ? "-c ${selectedContainer}" : ""} -- ps -ef`;

```

Comment:
Can kubectl output be structural by specifying some arguments? If so, please use that instead of parsing the human-readable output.

Reply:
`kubectl exec` command doesn't support `-o json` argument, `kubectl get` support that.